packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx

```diff
@@ -542,6 +539,20 @@ export const MetricsTable = () => {
         });
     }, [canManageTags, dataHasCategories, table]);
 
+    useEffect(
+        function handleRefetchOnViewChange() {
+            if (
+                data &&
+                metricCatalogView === MetricCatalogView.LIST &&
+                prevView.current === MetricCatalogView.TREE
+            ) {
+                void refetch();
```

Comment:
I think refetch is triggering a new request for the metrics

https://github.com/user-attachments/assets/23630763-378b-41e6-9b6c-ea9424cd09e9

Reply:
Perhaps we could try using `ConditionalVisibility` component?

Reply:
> I think refetch is triggering a new request for the metrics
yup! it will get the previous data - you just have the disabled cache checkbox enabled in dev mode

Reply:
>Perhaps we could try using ConditionalVisibility component?

will try!